src/server/auth/server/api_server.go

```diff
@@ -89,22 +74,14 @@ type apiServer struct {
 	txnEnv     *txnenv.TransactionEnv
 	pachLogger log.Logger
 
-	adminCache map[string]auth.ClusterRoles // cache of current cluster admins
-	adminMu    sync.Mutex                   // guard 'adminCache'
-
-	configCache *keycache.Cache
+	configCache             *keycache.Cache
+	clusterRoleBindingCache *keycache.Cache
```

Comment:
Use a keycache to cache the set of role bindings for the cluster, for performance.